In [33]:
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt 
import seaborn as sns
sns.set()
from scipy.stats import ttest_ind

In [34]:
df = pd.read_csv('marketing_AB.csv')

In [3]:
df.head()

Unnamed: 0  user id test group  converted  total ads most ads day  \
0           0  1069124         ad      False        130       Monday   
1           1  1119715         ad      False         93      Tuesday   
2           2  1144181         ad      False         21      Tuesday   
3           3  1435133         ad      False        355      Tuesday   
4           4  1015700         ad      False        276       Friday   

   most ads hour  
0             20  
1             22  
2             18  
3             10  
4             14

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 588101 entries, 0 to 588100
Data columns (total 7 columns):
 #   Column         Non-Null Count   Dtype 
---  ------         --------------   ----- 
 0   Unnamed: 0     588101 non-null  int64 
 1   user id        588101 non-null  int64 
 2   test group     588101 non-null  object
 3   converted      588101 non-null  bool  
 4   total ads      588101 non-null  int64 
 5   most ads day   588101 non-null  object
 6   most ads hour  588101 non-null  int64 
dtypes: bool(1), int64(4), object(2)
memory usage: 27.5+ MB


In [5]:
df.describe()

Unnamed: 0       user id      total ads  most ads hour
count  588101.000000  5.881010e+05  588101.000000  588101.000000
mean   294050.000000  1.310692e+06      24.820876      14.469061
std    169770.279667  2.022260e+05      43.715181       4.834634
min         0.000000  9.000000e+05       1.000000       0.000000
25%    147025.000000  1.143190e+06       4.000000      11.000000
50%    294050.000000  1.313725e+06      13.000000      14.000000
75%    441075.000000  1.484088e+06      27.000000      18.000000
max    588100.000000  1.654483e+06    2065.000000      23.000000

In [6]:
df.isnull().sum()

Unnamed: 0       0
user id          0
test group       0
converted        0
total ads        0
most ads day     0
most ads hour    0
dtype: int64

In [7]:
df.shape

(588101, 7)

In [8]:
df.columns

Index(['Unnamed: 0', 'user id', 'test group', 'converted', 'total ads',
       'most ads day', 'most ads hour'],
      dtype='object')

In [9]:
print(df['user id'].count())
print(df['total ads'].count())

588101
588101


In [10]:
print(df['test group'].unique())
print(df['converted'].unique())
print(df['most ads day'].unique())
print(df['most ads hour'].unique())

['ad' 'psa']
[False  True]
['Monday' 'Tuesday' 'Friday' 'Saturday' 'Wednesday' 'Sunday' 'Thursday']
[20 22 18 10 14 13 19 11 12 16 21  3 23  4  8  0  2 15  1  6 17  7  9  5]


In [11]:
#We will change the converted column to 1 or 0 (1 = True ; 0 = False)
df['converted']= df['converted'].apply(lambda x:1 if x==True else 0)

In [12]:
df.head()

Unnamed: 0  user id test group  converted  total ads most ads day  \
0           0  1069124         ad          0        130       Monday   
1           1  1119715         ad          0         93      Tuesday   
2           2  1144181         ad          0         21      Tuesday   
3           3  1435133         ad          0        355      Tuesday   
4           4  1015700         ad          0        276       Friday   

   most ads hour  
0             20  
1             22  
2             18  
3             10  
4             14

In [15]:
df.groupby('converted').count()['test group'].unique()

array([573258,  14843], dtype=int64)

In [16]:
df['test group'].unique()

array(['ad', 'psa'], dtype=object)

* 573,258 were converted in the ad group 
* 14,843 were converted in the psa group

In [22]:
df_shown = pd.DataFrame(df.groupby('test group')['total ads'].sum())
df_shown

total ads
test group           
ad           14014701
psa            582481

* ad was shown 14,014,701 times 
* psa was shown 582,481 times

In [32]:
df_convert = pd.DataFrame(df.groupby('test group')['converted'].mean())
df_convert

converted
test group           
ad           0.025547
psa          0.017854

* ad conversion rate = 2.5%
* psa conversion rate = 1.7%

In [47]:
ttest_report = ttest_ind(df[df['test group'] == 'ad']['converted'], df[df['test group'] == 'psa']['converted'])
ttest_report

Ttest_indResult(statistic=7.37040597428566, pvalue=1.7033052627831264e-13)

t-statistic indicates a significant difference between the means of ad and psa